### PR TITLE
chore(rhdh): add branch protection rules for...

### PR DESCRIPTION
**Pull request by nickboldt**
chore(rhdh): add branch protection rules for rhd/red-hat-developer-hub repo

Signed-off-by: Nick Boldt <nboldt@redhat.com>

### CODE REVIEW

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/_prowconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  orgs:
+    redhat-developer:
+      repos:
+        red-hat-developer-hub:
+          branches:
+            1.2.x:
+              protect: true
+              required_status_checks:
+                contexts:
+                - Main Job
```
Comment (schultzp2020) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/_prowconfig.yaml:11]: The action names need to be updated.

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/_prowconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                - PR Docker Build
+tide:
+  merge_method:
+    redhat-developer/red-hat-developer-hub: squash
+  queries:
+  - excludedBranches:
```
Comment (schultzp2020) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/_prowconfig.yaml:35]: We don't have any excluded branches.

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

**Comment by schultzp2020** [comment]
@nickboldt we'll need to copy [these changes](https://github.com/openshift/release/pull/58276).

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- davidfestal
+- debsmita1
+- divyanshigupta
+- gashcrumb
+- invinciblejai
+- josephca
+- kadel
+- kim-tsao
+- nickboldt
+- rm3l
```
Comment (nickboldt) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS:19]:
```suggestion
- rm3l
- schultz2020
```

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# DO NOT EDIT; this file is auto-generated using https://github.com/openshift/ci-tools.
+# Fetched from https://github.com/redhat-developer/red-hat-developer-hub root OWNERS
+# If the repo had OWNERS_ALIASES then the aliases were expanded
+# Logins who are not members of 'openshift' organization were filtered out
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
+
+approvers:
+- christoph-jerolimov
```
Comment (nickboldt) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS:8]:
```suggestion
- 04kash
- christoph-jerolimov
```

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+- coreydaley
+- davidfestal
+- debsmita1
+- divyanshigupta
+- gashcrumb
+- invinciblejai
+- josephca
```
Comment (nickboldt) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS:15]:
```suggestion
- JessicaJHee
- josephca
```

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- kadel
+- kim-tsao
+- nickboldt
+- rm3l
```
Comment (nickboldt) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS:19]:
```suggestion
- PatAKnight
- rm3l
```

File: core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+- schultz2020
+- subhashkhileri
+options: {}
+reviewers:
+- dzemanov
+- fortune-ndlovu
```
Comment (nickboldt) [inline core-services/prow/02_config/redhat-developer/red-hat-developer-hub/OWNERS:25]:
```suggestion
- Fortune-Ndlovu
- its-mitesh-kumar
```

**Pull request merged**